spark-publish/index.js

```diff
@@ -66,21 +66,30 @@ export const publish = async ({
   const ieAddMeasurementsDuration = new Date() - start
   logger.log('Measurements added to round', roundIndex.toString())
 
-  // Mark measurements as shared
-  await client.query(`
-    UPDATE measurements
-    SET published_as = $1
-    WHERE id = ANY($2::int[])
-  `, [
-    cid.toString(),
-    measurements.map(m => m.id)
-  ])
+  try {
+    await client.query('BEGIN')
```

Comment:
This does not work. See https://node-postgres.com/features/transactions

> You must use the same client instance for all statements within a transaction. PostgreSQL isolates a transaction to individual clients. This means if you initialize or use transactions with the pool.query method you will have problems. Do not use transactions with the pool.query method.

This is how I am have solved this problem elsewhere:
https://github.com/filecoin-station/spark-api/blob/ca10df39450cf50ff68be871869b763287bc0939/lib/round-tracker.js#L33-L47

I would like to eventually introduce a helper function for running queries in a transaction, but maybe it's too early if we need it in two places only.

The idea:

```js
// usage
await runInTransaction(pgPool, (client) => {
 await client.query()
 await client.query()
})

// implementation
const runInTransaction = (pgPool, fn) => {
  const pgClient = await pgPool.connect()
  try {
    await pgClient.query('BEGIN')
    await fn(pgClient)
    await pgClient.query('COMMIT')
   } catch (err) {
    await pgClient.query('ROLLBACK')
    throw err
   } finally {
     pgClient.release()
  }
}

Reply:
Unless we can be sure that `client` is always `pg.Client` and never `pg.Pool`?

Reply:
We're passing a pg pool here indeed. 

> I would like to eventually introduce a helper function for running queries in a transaction, but maybe it's too early if we need it in two places only.

I was wondering about this too, it seems strange that `pg` doesn't offer this. I think it would be nice to introduce this